collector/os_release.go

```diff
@@ -128,6 +129,11 @@ func parseOSRelease(r io.Reader) (*osRelease, error) {
 }
 
 func (c *osReleaseCollector) UpdateStruct(path string) error {
+	path, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return err
+	}
+
```

Comment:
I'm still not sure why the open/stat isn't following the symlink on your system. I am unable to reproduce this on Ubuntu.

Would you mind adding a `Debug` here so we can log the final path?

```suggestion
	level.Debug(c.logger).Log("msg", "Opening os-release file", "file", path)
```